[MOIA-30005] Migrate to GitHub Actions

m-dobler opened the pull request:
Adds a GitHub actions workflow to replace the existing circleCI workflow defined [here](https://github.com/moia-dev/teleproto/blob/master/.circleci/config.yml). The workflow just consists of a job that runs the tests.

If this should get merged, some admin of this circleCI project should disable builds in the [project settings](https://app.circleci.com/settings/project/github/moia-dev/teleproto?return-to=https%3A%2F%2Fapp.circleci.com%2Fpipelines%2Fgithub%2Fmoia-dev%2Fteleproto), I do not have the rights for it. 

#### Has the version number been increased?
 - [x] No, but it was intended not to increase it

m-dobler commented on .github/workflows/run-tests.yml:
```diff
@@ -0,0 +1,18 @@
+name: Run tests
+on: push
```
The circleCI workflow runs on every branch except `master`, but I am not sure if it should really be excluded from this simple test workflow.

aquamatthias replied:
Testing on master would be too late and should be done before.
There is no harm in executing tests on master again.

m-dobler commented on .github/workflows/run-tests.yml:
```diff
@@ -0,0 +1,18 @@
+            ~/.sbt
+          key: ${{ runner.os }}-sbt-ivy-cache-${{ hashFiles('**/build.sbt') }}-${{ hashFiles('**/plugins.sbt') }}-${{ hashFiles('**/build.properties') }}
+      - name: Compile and test
+        run: sbt +test
```
The circleCI workflow executes the command 
```
sbt -J-Xms512m -J-Xmx2g -J-XX:+UseConcMarkSweepGC -J-XX:ReservedCodeCacheSize=256m -J-XX:+CMSClassUnloadingEnabled -Djava.awt.headless=true -no-colors -batch "+test"
```
I left out all of these flags just to keep it simple, but wouldn't mind adding some/all back.

aquamatthias replied:
The flags have been added on purpose, so I assume they were needed in the past.
Consider to add the options into the `.sbtopts` file to have a clean separation.

aquamatthias replied:
Please also add scapegoat check.

m-dobler replied:
> Consider to add the options into the .sbtopts file to have a clean separation.

Alright, done in c9a7d0f

> Please also add scapegoat check

Scapegoat job was added. It failed with two errors because of variable shadowing, which I fixed in 1848a8e (but I actually prefer how it was previously written)

aquamatthias replied:
I would suggest to disable this inspection via this sbt setting:
```
scapegoatDisabledInspections := Seq("VariableShadowing")
```

m-dobler replied:
Good idea, thanks! Disabled this inspection and reverted the previous code changes in 44d4315

m-dobler commented:
> Thanks. We also should remove circleCI config?

Yes, I removed the config now 66f196e. But the project is still building and failing because of a missing config. So the builds would have to now be stopped in the [circleCI project settings](https://app.circleci.com/settings/project/github/moia-dev/teleproto?return-to=https%3A%2F%2Fapp.circleci.com%2Fpipelines%2Fgithub%2Fmoia-dev%2Fteleproto). I don't have the required rights for that, maybe you do @aquamatthias ?

I now also changed the required PR status checks to include the GitHub Actions workflows.
